js/preferences/Preferences.js

```diff
@@ -406,6 +595,38 @@ export async function __setup__() {
         ConstructorControlPanel.showDebug('Failed to load LTS runtimes list', lts_res.err);
     }
 
+    if (user_stable_res.ok) {
+        users.Stable = user_stable_res.data;
+    } else {
+        ConstructorControlPanel.showDebug('Failed to load Stable users list', user_stable_res.err);
+    }
+
+    if (user_beta_res.ok) {
+        users.Beta = user_beta_res.data;
+    } else {
+        ConstructorControlPanel.showDebug('Failed to load Beta users list', user_beta_res.err);
+    }
+
+    if (user_lts_res.ok) {
+        users.LTS = user_lts_res.data;
+    } else {
+        ConstructorControlPanel.showDebug('Failed to load LTS users list', user_lts_res.err);
+    }
+
+    // If some runtimes/users aren't filled in but there is an available runtime version/user,
+    // fill them in automatically
+    for (const runtime_type of valid_runtime_types) {
+        const options = prefs.runtime_opts.type_opts[runtime_type];
+        if (!options.choice && runtimes[runtime_type] && (runtimes[runtime_type]?.length ?? 0) > 0) {
+            // @ts-ignore
+            options.choice = runtimes[runtime_type][0].version.toString();
+        }
+        if (!options.user && users[runtime_type] && (users[runtime_type]?.length ?? 0) > 0) {
+            // @ts-ignore
+            options.user = users[runtime_type][0].name.toString();
+        }
+    }
+
```

Comment:
This one when I did it was already on the verge of needing to be switched out for a nicer `for...of [Stable, Beta, LTS]...` type thing, I think at this point it definitely needs that lol.